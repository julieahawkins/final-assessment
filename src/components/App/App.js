import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { fetchData } from '../../actions';
class App extends Component {
  componentDidMount() {
    // this.props.fetchData();
    // alert(this.props.fake);
    
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
          <button onClick={() => {
            this.props.fetchData('data');
          }}> Fetch </button>
        </div>
        <div className='Display-info'>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  data: shape({ data: string }),
  fetchData: func.isRequired
};

const mapStateToProps = ({ data }) => ({ data });
const mapDispatchToProps = dispatch => ({ 
  fetchData: (data) => dispatch(fetchData(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
