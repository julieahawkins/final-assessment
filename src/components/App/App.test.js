import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import configureMockStore from 'redux-mock-store';


describe('App test', () => {
  const fakeStore = configureMockStore()({
    data: []
  });

  const setup = () => {
    const props = {
      fetchData: jest.fn(),
    }
   
    const wrapper = shallow(
      <Provider store={fakeStore}>
        <App />
      </Provider>
    );

    const Component = wrapper.find(App);

    return {
      props,
      Component
    };
  }  


  it('should match its snap shot', () => {

    expect(setup.wrapper).toMatchSnapshot();
  });
})