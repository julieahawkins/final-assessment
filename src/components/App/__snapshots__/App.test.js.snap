// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should match its snap shot 1`] = `undefined`;
